pkg/applicationserver/io/pubsub/unique/unique.go

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unique
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var uniqueIDPattern = regexp.MustCompile("(.*)\\:(.*)")
+
+// ID generates an unique ID based on an application unique ID and a PubSub ID.
+func ID(appUID, id string) string {
```

Comment:
Hmm we only use this in the Redis registry, right? Can we put this there instead?

This is going to become confusing with `pkg/unique` I'm afraid

Reply:
Both Redis and the main package use this (the main package uses it for the `sync.Map`). Another name perhaps ?

Reply:
I see. Does it deserve its own package then?

Maybe `PubsubUID(context.Context, ttnpb.ApplicationIdentifiers, string) string` and `SplitPubsubUID() (string, string)`?

Reply:
It's a bit too small for a separate package, but I wanted to keep cyclic dependencies at bay. I'll move them to the main package, renamed....................